hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/flow/dispatch/logic/DispatchProcessor.java

```diff
@@ -220,17 +220,17 @@ private void chargeCreator(@NonNull final Dispatch dispatch, @NonNull final Erro
      * @param report the due diligence report for the dispatch
      */
     private void chargePayer(@NonNull final Dispatch dispatch, @NonNull final ErrorReport report) {
-        final var fees = dispatch.fees();
-        if (fees.nothingToCharge()) {
-            return;
-        }
         final var hasWaivedFees = authorizer.hasWaivedFees(
                 dispatch.payerId(),
                 dispatch.txnInfo().functionality(),
                 dispatch.txnInfo().txBody());
         if (hasWaivedFees) {
             return;
         }
+        final var fees = dispatch.fees();
+        if (fees.nothingToCharge()) {
+            return;
+        }
```

Comment:
This change looks like something that should be in a separate PR and not part of updating a design document

Reply:
Reverted. Thanks @rbair23